### PR TITLE
feat: Support Azure single-tenant application using the Graph API (#6728)

### PR DESCRIPTION
**Pull request by tejaswinichile**
Fixes: https://linear.app/chatwoot/issue/CW-1400/getting-microsoft-email-working-with-chatwoot-in-2023

### CODE REVIEW

File: app/jobs/inboxes/fetch_imap_email_inboxes_job.rb [hunk 1]
```diff
@@ -2,8 +2,19 @@ class Inboxes::FetchImapEmailInboxesJob < ApplicationJob
...
-      ::Inboxes::FetchImapEmailsJob.perform_later(inbox.channel) if inbox.channel.imap_enabled
+      next unless inbox.channel.imap_enabled?
+
+      fetch_emails(inbox)
+    end
+  end
+
+  def fetch_emails(inbox)
+    if inbox.channel.microsoft? && ENV.fetch('AZURE_TENANT_ID', false)
+      ::Inboxes::FetchMsGraphEmailsJob.perform_later(inbox.channel)
```
Comment (sojan-official) [inline app/jobs/inboxes/fetch_imap_email_inboxes_job.rb:15]: `fetchMsGraphEmailForTenantJob` ?  in the name to depicts this works on for the case of single tenant ?

File: app/jobs/inboxes/fetch_ms_graph_emails_job.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
+require 'net/http'
+
+class Inboxes::FetchMsGraphEmailsJob < ApplicationJob
```
Comment (sojan-official) [inline app/jobs/inboxes/fetch_ms_graph_emails_job.rb:3]: prev comment. name that denotes single clients

File: app/jobs/inboxes/fetch_ms_graph_emails_job.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  queue_as :low
```
Comment (sojan-official) [inline app/jobs/inboxes/fetch_ms_graph_emails_job.rb:4]: lets queue this on scheduled_jobs queue

**Review by sojan-official — COMMENTED** [review]
Minor comments. also, lets add some specs for 
- authorization controller 
- reply mailer
- lib/microsoft_graph_api  ( get from API and post to API methods )

also there seems to be some build issues

File: spec/jobs/inboxes/fetch_imap_email_inboxes_job_spec.rb [hunk 4]
```diff
@@ -20,4 +26,24 @@
       described_class.perform_now
     end
   end
+
+  context 'when microsoft inbox' do
+    it 'calls ms graph channel for single tenant app' do
```
Comment (sojan-official) [inline spec/jobs/inboxes/fetch_imap_email_inboxes_job_spec.rb:31]:
```suggestion
    it 'calls fetch ms graph email job for single tenant app' do
```

**Comment by sojan-official** [comment]
note : @sojan-official to take a look into the issue during staging deploy

**Comment by sojan-official** [comment]
note: the issue was with autoloading.  verified that the server stars in RAILS_ENV=production

**Comment by tejaswinichile** [comment]
Update documentation with this link: 
https://learn.microsoft.com/en-us/graph/api/user-list-messages?view=graph-rest-1.0&tabs=http#code-try-2
And all new screenshots for single tenant app.

**Comment by tejaswinichile** [comment]
@sojan-official I have found one issue in single tenant changes, that's if the same user have one multi-tenant app and one single tenant app, the microsoft-grap call is not getting executed. I don't know how relevant is this could be for the real time user. 
We can merge this by keeping ENV['AZURE_TENANT_ID'] check to false, so single tenant changes will never get called till we fix the issue.
Else we can icebox this one, because I spent 2 hours for this to find the real cause and I did not find any.

**Comment by sojan-official** [comment]
We need to make sure
1. if an account has an approved app  and an unapproved app. this wouldn't break for the approved app. ( works without tenant id )
2. we need to make sure for an account with just an unapproved app (it works with the tenant id)

-------
We won't be looking into the case where the client has both approved and unapproved, and then they try to configure with the unapproved app for now.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**